syft/formats/syftjson/decoder.go

```diff
@@ -18,5 +22,26 @@ func decoder(reader io.Reader) (*sbom.SBOM, error) {
 		return nil, fmt.Errorf("unable to decode syft-json: %w", err)
 	}
 
+	if err := outOfDateParser(doc.Schema.Version, internal.JSONSchemaVersion); err != nil {
+		log.Warn(err)
+	}
+
 	return toSyftModel(doc)
 }
+
+func outOfDateParser(documentVerion string, parserVersion string) error {
```

Comment:
nit: naming-wise this is really asserting/checking/validating the date, so something like `checkSupportedSchema` or similar seems a little better.